pkg/trace/writer/trace.go

```diff
@@ -49,32 +50,37 @@ type TraceWriter struct {
 	// Channel should only be received from when testing.
 	In chan *SampledSpans
 
-	hostname string
-	env      string
-	senders  []*sender
-	stop     chan struct{}
-	stats    *info.TraceWriterInfo
-	wg       sync.WaitGroup // waits for gzippers
-	tick     time.Duration  // flush frequency
+	hostname  string
+	env       string
+	senders   []*sender
+	stop      chan struct{}
+	flushChan chan chan struct{}
+	stats     *info.TraceWriterInfo
+	wg        sync.WaitGroup // waits for gzippers
+	tick      time.Duration  // flush frequency
 
 	traces       []*pb.APITrace // traces buffered
 	events       []*pb.Span     // events buffered
 	bufferedSize int            // estimated buffer size
+	syncMode     bool           // when enabled, traces are only sent when Flush is called
```

Comment:
```suggestion
	syncMode     bool           // when enabled, traces are only sent when FlushSync is called
```
Would you mind creating a separate group of fields (just like we did in the stats writer) with the fields that are used only in sync mode? I mean `syncMode` and `flushChan`